File: frontend/src/utils/validation.ts
export const validateEmail = (email: string): boolean => {
  const re = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
  return re.test(email);
};

export const validatePassword = (password: string): boolean => {
  // At least 8 characters, one uppercase, one lowercase, one number
  const re = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)[a-zA-Z\d@$!%*?&]{8,}$/;
  return re.test(password);
};

export const validatePhoneNumber = (phone: string): boolean => {
  // Basic phone number validation
  const re = /^\+?[\d\s\-\(\)]{10,}$/;
  return re.test(phone);
};

export const validateAccountNumber = (accountNumber: string): boolean => {
  // Basic account number validation (alphanumeric, 8-20 characters)
  const re = /^[A-Za-z0-9]{8,20}$/;
  return re.test(accountNumber);
};

export const validateAmount = (amount: string): boolean => {
  const num = parseFloat(amount);
  return !isNaN(num) && num > 0;
};

File: backend/internal/utils/validation.go
package utils

import (
	"regexp"
	"strings"
)

// ValidateEmail validates an email address
func ValidateEmail(email string) bool {
	// Basic email validation regex
	emailRegex := regexp.MustCompile(`^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$`)
	return emailRegex.MatchString(email)
}

// ValidatePassword validates a password
func ValidatePassword(password string) bool {
	// Password should be at least 8 characters long
	return len(password) >= 8
}

// ValidatePhoneNumber validates a phone number
func ValidatePhoneNumber(phoneNumber string) bool {
	// Basic phone number validation (allows + and digits)
	phoneRegex := regexp.MustCompile(`^\+?[1-9]\d{1,14}$`)
	return phoneRegex.MatchString(phoneNumber)
}

// ValidateCurrencyCode validates a currency code
func ValidateCurrencyCode(currencyCode string) bool {
	// Currency code should be 3 uppercase letters
	currencyRegex := regexp.MustCompile(`^[A-Z]{3}$`)
	return currencyRegex.MatchString(strings.ToUpper(currencyCode))
}

// ValidateCountryCode validates a country code
func ValidateCountryCode(countryCode string) bool {
	// Country code should be 2 uppercase letters
	countryRegex := regexp.MustCompile(`^[A-Z]{2}$`)
	return countryRegex.MatchString(strings.ToUpper(countryCode))
}